pkg/git.go

```diff
@@ -36,16 +35,39 @@ func NewGitPackage(source *spec.GitSource) Interface {
 }
 
 func (p *GitPackage) Install(ctx context.Context, dir, version string) (lockVersion string, err error) {
-	cmd := exec.CommandContext(ctx, "git", "clone", p.Source.Remote, dir)
-	cmd.Stdin = os.Stdin
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	err = cmd.Run()
-	if err != nil {
-		return "", err
+	_, err = os.Stat(dir)
+	if err != nil && os.IsNotExist(err) {
+		cmd := exec.CommandContext(ctx, "git", "clone", p.Source.Remote, dir)
+		cmd.Stdin = os.Stdin
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		err = cmd.Run()
+		if err != nil {
+			return "", err
+		}
+	} else {
```

Comment:
You should be able to avoid this else by returning in the first if part.

Reply:
Ah but we still need the commands below to make sure we checkout the right branch etc.

Reply:
Yes, you can simply indent them to the left once. That should be still the same, right?

Reply:
I don't follow: are you saying we should always do the remote set-url and fetch?  I guess we can, they would just be redundant.

Reply:
I don't think we can indent this; the first two if blocks clone a fresh repo or fetch updates to the existing repo, the rest of the body checks out the right version.